sensorprocess/sensorprocess.go

```diff
@@ -12,39 +11,85 @@ import (
 	s "github.com/viamrobotics/viam-cartographer/sensors"
 )
 
-// defaultTime is used to check if timestamps have not been set yet.
-var defaultTime = time.Time{}
-
 // Config holds config needed throughout the process of adding a sensor reading to the cartofacade.
 type Config struct {
-	CartoFacade            cartofacade.Interface
-	sensorProcessStartTime time.Time
-	IsOnline               bool
-
-	Lidar            s.TimedLidar
-	currentLidarData *s.TimedLidarReadingResponse
+	CartoFacade cartofacade.Interface
+	IsOnline    bool
 
-	IMU            s.TimedMovementSensor
-	currentIMUData *s.TimedIMUReadingResponse
+	Lidar s.TimedLidar
+	IMU   s.TimedMovementSensor
 
 	Timeout                  time.Duration
 	InternalTimeout          time.Duration
 	Logger                   logging.Logger
 	RunFinalOptimizationFunc func(context.Context, time.Duration) error
```

Comment:
This was so that we could unit test

Reply:
the timeout exists because the call to RunFinalOptimization takes longer than time out used for the rest of our APIs. We wanted to create an internal timeout to represent the calls we make that we expect to take a longer time

Reply:
I believe we can use cartofacade's `RunFinalOptimization` directly, unless there is a reason I can't think of why we should not do this. I made the change and am pushing it - please review it if it looks correct.